modules/dynamic-plugin-proxy-service.adoc

```diff
@@ -0,0 +1,43 @@
+// Module included in the following assemblies:
+//
+// * web_console/dynamic-plugin/deploy-plugin-cluster.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="dynamic-plugin-proxy-service_{context}"]
+= Plugin service proxy
+
+You can declare a service proxy in its `ConsolePlugin` resource by using the `spec.proxy` array field if you need to make HTTP requests to an in-cluster service with your plugin. The console backend exposes the `/api/proxy/plugin/<plugin-name>/<proxy-alias>/<request-path>++?++<optional-query-parameters>` endpoint to proxy the communication between plugin and the service. A proxied request uses a _service CA bundle_ by default. The service must use HTTPS.
+
+[NOTE]
+====
+The plugin must have the `consolefetch` API to make requests from its JavaScript code or some requests might fail.
+====
+
+For each entry you must specify an endpoint and alias of the proxy under the `endpoint` and `alias` fields. For the Service proxy type, the endpoint `type` field will need to be set to `Service` and the `service` must include a `name`, `namespace`, and `port`. For example, `/api/proxy/plugin/helm/helm-charts/releases++?++limit++=++10` is a proxy request path from the `helm` plugin with a `helm-charts` service that lists ten helm releases.
+
+.Example
+[source,text,subs="+quotes,+macros"]
+----
```

Comment:
Even though you're showing the `spec`, should include the [required API fields](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#required-fields) per docs guidelines. I am guessing this is not global, so probably just need the `apiVersion` and `kind` here.

Reply:
> Even though you're showing the `spec`, should include the [required API fields](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#required-fields) per docs guidelines. I am guessing this is not global, so probably just need the `apiVersion` and `kind` here.

You'd want `metadata.name` as well